### PR TITLE
Add autocompletion for type flag

### LINKED ISSUE #360: Tab completion doesn't work for `--type` flag values
- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
Tab completion works for flag values, but doesn't work for `--type`

### Expected behavior
When I hit <TAB> after typing `tanzu apps workload create/apply APP-NAME --type`
I should see the value(s) for all of the workload types that are supported on the target cluster

### Steps to Reproduce
1. target a cluster with supply chain(s) that support multiple workload types
2. type `tanzu apps workload create/apply APP-NAME --type ` (include a space after "--type"
3. hit <TAB>

### Version (Apps plugin version, Version of K8s running on cluster)
Put the output of the following commad
```sh
tanzu version && tanzu apps version
``` 
version: v0.27.0-dev
buildDate: 2022-09-13
sha: 0b58fbea
v0.9.0+dev-baf0aa02

### Environment where the bug was observed (cloud, OS, etc)
This issue occurs regardless of which supply chain versions are running on the target cluster

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
Add autocompletion func for `--type` flag. 

### Which issue(s) this PR fixes

Fixes #360 

### Describe testing done for PR

- Modified OOTB supply chain to have different selectors besides `web`, `server`, `worker` and showed also the custom ones.

### CODE REVIEW

File: pkg/commands/workload.go [hunk 1]
```diff
@@ -792,13 +794,35 @@ func (opts *WorkloadOptions) GetTailWorkers(c *cli.Config, workload *cartov1alph
 	return workers
 }
 
+func getClusterSupplyChainTypeSelectors(fields []metav1.LabelSelectorRequirement) []string {
+	var values []string
+	for _, v := range fields {
+		if v.Key == apis.WorkloadTypeLabelName && len(v.Values) != 0 {
```
Comment (odinnordico) [inline pkg/commands/workload.go:800]: `len` check is not required, see https://go.dev/play/p/7tJe1k_QO_v

Comment (warango4) [reply pkg/commands/workload.go]: ok, removed `len` validation

**Review by odinnordico — APPROVED** [review]

File: pkg/commands/workload.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ import (
 
 	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (shaheerkootteeri) [inline pkg/commands/workload.go:44]: Please group dependencies as below 
Format:

```

Standard library packages

External packages 

Local source code packages

```

Comment (warango4) [reply pkg/commands/workload.go]: Updated

**Review by shaheerkootteeri — APPROVED** [review]

**Review by odinnordico — APPROVED** [review]

**Pull request merged**